### DISCUSSION TITLE
NanoMDM multi tenancy

### DISCUSSION THREAD

**Discussion by paolominel** [micromdm/nanomdm#76] [Q&A] [answered]
Is it possible to configure a multi-tenancy nanoMDM solution?

Can I use:
- a single instance a SCEP server (with all the certificates and the keys for the registered APNs)
- a single instance of a nanoMDM with many endpoint (one enpoint for one customer). A single endpoint must be in format like ".../**customer_name**/mdm/...."

Is it the right way to achieve a multi-tenancy nanoMDM configuration?

**Comment by jessepeterson** [comment, marked as answer]
Hello! 

> a single instance a SCEP server (with all the certificates and the keys for the registered APNs)

You can use a single SCEP server with NanoMDM (or multiple NanoMDM instances if you wanted) yes. However SCEP doesn't have much to do with APNs — those certificates are obtained from Apple via https://identity.apple.com/.

> a single instance of a nanoMDM with many endpoint (one enpoint for one customer). A single endpoint must be in format like ".../customer_name/mdm/...."

Yes, NanoMDM supports using multiple APNs push *topics* (i.e. multiple APNs certificates). You can target them however you wish, though it's probably easiest to use the URL parameters (as these are passed through MDM and into the web hook callbacks). So it'd be something like `.../mdm?customer=name` instead. Note though that this URL can never change for the lifetime of an enrollment. Also make sure the `Topic` key in the enrollment profile refers to the correct APNs certificate/topic.

Hope that helps!
